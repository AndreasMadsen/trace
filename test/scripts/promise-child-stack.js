'use strict';

const p0 = Promise.resolve(1); // will have `scope` in stack trace
const p1 = p0.then(function scopeThrow() {
  console.error(new Error('hello').stack);
});
