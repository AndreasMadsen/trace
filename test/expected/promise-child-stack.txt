Error: hello
    at scopeThrow (/scripts/promise-child-stack.js:r:c)
    at process._tickCallback (internal/process/next_tick.js:r:c)
    at Function.Module.runMain (internal/modules/cjs/loader.js:r:c)
    at startup (internal/bootstrap/node.js:r:c)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:r:c)
    at Promise.then (<anonymous>)
    at Object.<anonymous> (/scripts/promise-child-stack.js:r:c)
    at Function.resolve (<anonymous>)
    at Object.<anonymous> (/scripts/promise-child-stack.js:r:c)
    at Module._compile (internal/modules/cjs/loader.js:r:c)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:r:c)
    at Module.load (internal/modules/cjs/loader.js:r:c)
    at tryModuleLoad (internal/modules/cjs/loader.js:r:c)
    at Function.Module._load (internal/modules/cjs/loader.js:r:c)
    at Function.Module.runMain (internal/modules/cjs/loader.js:r:c)
    at startup (internal/bootstrap/node.js:r:c)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:r:c)
